File: PassDataBackwardsObjC/PassDataBackwardsObjC/SecondaryViewController.m
//
//  SecondaryViewController.m
//  PassDataBackwardsObjC
//
//  Created by Field Employee on 4/2/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

#import "SecondaryViewController.h"

@interface SecondaryViewController ()

@end

@implementation SecondaryViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)onPress:(id)sender {
    [_mainViewController onUserAction:@"Hello From Second"];
    [self.navigationController popViewControllerAnimated:true];
    [self dismissViewControllerAnimated:true completion:nil];
}

@end

File: PassDataBackwardsSwift/PassDataBackwardsSwift/SecondaryViewController.swift
//
//  SecondaryViewController.swift
//  PassDataBackwardsSwift
//
//  Created by Field Employee on 4/2/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

import UIKit

class SecondaryViewController: UIViewController {
    
    var mainViewController: ViewController?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    @IBAction func onPress(_ sender: Any) {
        mainViewController?.onUserAction(data: "Hello From Second")
        navigationController?.popViewController(animated: true)
        dismiss(animated: true, completion: nil)
    }
    
}
